internal/corazarules/rule_match.go

```diff
@@ -132,10 +134,22 @@ func (mr MatchedRule) writeDetails(log *strings.Builder, matchData types.MatchDa
 		mr.ServerIPAddress_, mr.URI_, mr.TransactionID_))
 }
 
-func (mr MatchedRule) matchData(matchData types.MatchData, log *strings.Builder) {
+func (mr MatchedRule) writeExtraRuleDetails(log *strings.Builder, matchData types.MatchData, n int) {
+	msg := matchData.Message()
+	data := matchData.Data()
+	if len(msg) > maxSizeLogMessage {
+		msg = msg[:maxSizeLogMessage]
+	}
+	if len(data) > maxSizeLogMessage {
+		data = data[:maxSizeLogMessage]
+	}
+	log.WriteString(fmt.Sprintf("[msg_match_%d %q] [data_match_%d %q]", n, msg, n, data))
```

Comment:
If you want to use printf with strings.Builder, I think you can still avoid the extra allocation with something like

`fmt.Fprintf(log, "[msg_match_%d %q] [data_match_%d %q]", n, msg, n, data)`

Applies throughout the PR

Reply:
Also it's a bit surprising to be printing the same value `n` twice